src/hooks/useLocalAndRemoteHistory.js

```diff
@@ -23,11 +23,27 @@ const reducer = (state, action) => {
       const { payload } = action;
       const { data } = payload;
       const { data: initialData } = state;
-      const localIds = initialData.map(history => history.id);
+      const localTransactionIds = initialData.map(transactions => transactions.id);
+
+      // Name notes do not have transaction ids, create filter instead based on itemCode/confirmDate
```

Comment:
This has a big assumption that someone can't receive two doses of the same vaccine in the same day to do duplicate detection between local/remote transactions 🤔  - not sure if there is a better way to filter.

Reply:
there is `extra.prescription.id` and we can add the id to the name note similar to doses.

So `id = nameNote.id || extra.prescription.id` ? Bit yuck but

Reply:
oops more like.. `id = nameNote.transactionId || nameNote.extra.prescription.id` - hopefully you get the idea .. 😛

Reply:
Ooh, I see what you mean, the whole prescription is supposed to be in the nameNote 😮  that would make life a lot easier - I think at the moment it's not (or at least not on my environment unless I've done something weird)...

Prescription gets created first, and then theoretically the name note should take the prescription as an arg:
https://github.com/openmsupply/mobile/blob/75d8f47cdfc5ce78786b29f4e5d8fc326a36152d/src/actions/Entities/VaccinePrescriptionActions.js#L273-L289

Except it looks like `createPrescription` doesn't actually return the prescription?
https://github.com/openmsupply/mobile/blob/75d8f47cdfc5ce78786b29f4e5d8fc326a36152d/src/actions/Entities/VaccinePrescriptionActions.js#L145-L171